rust-runtime/aws-smithy-http-server/src/protocols.rs

```diff
@@ -20,35 +20,76 @@ pub struct AwsJson10;
 pub struct AwsJson11;
 
 /// Supported protocols.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, PartialEq)]
 pub enum Protocol {
     RestJson1,
     RestXml,
     AwsJson10,
     AwsJson11,
 }
 
-pub fn check_content_type<B>(
+pub fn content_type_header_empty_body_no_modeled_input<B>(
     req: &RequestParts<B>,
-    expected_mime: &'static mime::Mime,
+    empty_body: bool,
+    protocol: Protocol,
 ) -> Result<(), MissingContentTypeReason> {
-    let found_mime = req
-        .headers()
-        .ok_or(MissingContentTypeReason::HeadersTakenByAnotherExtractor)?
+    if req.headers().is_none() || (protocol != Protocol::RestJson1 && protocol != Protocol::RestXml) {
```

Comment:
No, it's just that a client should always send an empty JSON object payload in AwsJson 1.0, so with `{}` in the body this would fail (this only checks whether the body is empty)